src/api/hooks/useWeatherApi.ts

```diff
@@ -0,0 +1,27 @@
+import { useEffect, useState } from "react";
+import { City, fetchWeatherData } from "../weather";
+
+export const useWeatherApi = () => {
+  const [cities, setCities] = useState<City[]>([]);
+  const [loading, setLoading] = useState<boolean>(true);
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    const fetchData = async () => {
```

Comment:
NIT: This looks awesome, just to note this could also be done as a IIFE, or an external function with a setter passed in. E.G:

```ts
// external function
const fetchData = async (setCities: (City[]) => void, setError: (err: string) => void, setLoading: (status: boolean) => void): Promise<void> => {
 ...
}

export const useWeatherApi = () => {
...
useEffect(() => {
  fetchData(setCities, setError, setLoading);
}, []);

...
}

```

```ts
// IIFE

export const useWeatherApi = () => {
...
useEffect(() => {
  (async () => {
    ...
  })();
}, []);

...
}